linkerd/proxy/transport/src/orig_dst.rs

```diff
@@ -100,6 +101,16 @@ where
 
         Ok((addr, Box::pin(incoming)))
     }
+
+    fn bind_additional(self, t: &T) -> Result<Bound<Self::Incoming>> {
+        match t.param() {
+            Some(addr) => {
+                let t2 = t.insert_param(addr, ());
+                self.bind(&t2)
+            }
+            None => self.bind(t),
+        }
+    }
```

Comment:
Instead, we can provide a new type, e.g. `DualBind<B>(B, B)` that binds over a `T: Param<DualListenAddr>`, and then when we construct the app, we give it a DualBind for the outbound listener instead!

The trick to this will be joining the two resultant Incoming streams, but we should have utilities available to us to accomplish that.